libsolidity/formal/CVC4Interface.cpp

```diff
@@ -129,20 +113,19 @@ pair<CheckResult, vector<string>> CVC4Interface::check(vector<Expression> const&
 
 CVC4::Expr CVC4Interface::toCVC4Expr(Expression const& _expr)
 {
-	if (_expr.arguments.empty() && m_constants.count(_expr.name))
-		return m_constants.at(_expr.name);
+	// Variable
+	if (_expr.arguments.empty() && m_variables.count(_expr.name))
+		return m_variables.at(_expr.name);
+
 	vector<CVC4::Expr> arguments;
 	for (auto const& arg: _expr.arguments)
 		arguments.push_back(toCVC4Expr(arg));
 
 	string const& n = _expr.name;
-	if (m_functions.count(n))
-		return m_context.mkExpr(CVC4::kind::APPLY_UF, m_functions[n], arguments);
-	else if (m_constants.count(n))
-	{
-		solAssert(arguments.empty(), "");
-		return m_constants.at(n);
-	}
+	// Function application
+	if (!arguments.empty() && m_variables.count(_expr.name))
```

Comment:
Instead of checking `arguments.empty`, should this function check `_expr.sort.kind` ?

Reply:
Not in this function, since what ends up here are not the functions themselves (`FunctionSort` represents function declarations), but function applications, which will have the `kind` of the `codomain` of the `FunctionSort`, or primitive ones for the built-in functions (eq, neq, and, or, etc).

): they are simply function declarations, and the application of functions